patches/net/minecraft/world/level/block/entity/ChestBlockEntity.java.patch

```diff
@@ -1,46 +1,14 @@
 --- a/net/minecraft/world/level/block/entity/ChestBlockEntity.java
 +++ b/net/minecraft/world/level/block/entity/ChestBlockEntity.java
-@@ -172,6 +_,45 @@
+@@ -172,6 +_,13 @@
        return ChestMenu.threeRows(p_59082_, p_59083_, this);
     }
  
-+   private net.neoforged.neoforge.common.util.LazyOptional<net.neoforged.neoforge.items.IItemHandlerModifiable> chestHandler;
 +   @Override
 +   public void setBlockState(BlockState p_155251_) {
 +      super.setBlockState(p_155251_);
-+      if (this.chestHandler != null) {
-+         net.neoforged.neoforge.common.util.LazyOptional<?> oldHandler = this.chestHandler;
-+         this.chestHandler = null;
-+         oldHandler.invalidate();
-+      }
-+   }
-+
-+   @Override
-+   public <T> net.neoforged.neoforge.common.util.LazyOptional<T> getCapability(net.neoforged.neoforge.common.capabilities.Capability<T> cap, Direction side) {
-+       if (!this.remove && cap == net.neoforged.neoforge.common.capabilities.Capabilities.ITEM_HANDLER) {
-+          if (this.chestHandler == null)
-+             this.chestHandler = net.neoforged.neoforge.common.util.LazyOptional.of(this::createHandler);
-+          return this.chestHandler.cast();
-+       }
-+       return super.getCapability(cap, side);
-+   }
-+
-+   private net.neoforged.neoforge.items.IItemHandlerModifiable createHandler() {
-+      BlockState state = this.getBlockState();
-+      if (!(state.getBlock() instanceof ChestBlock)) {
-+         return new net.neoforged.neoforge.items.wrapper.InvWrapper(this);
-+      }
-+      Container inv = ChestBlock.getContainer((ChestBlock) state.getBlock(), state, getLevel(), getBlockPos(), true);
-+      return new net.neoforged.neoforge.items.wrapper.InvWrapper(inv == null ? this : inv);
-+   }
-+
-+   @Override
-+   public void invalidateCaps() {
-+      super.invalidateCaps();
-+      if (chestHandler != null) {
-+         chestHandler.invalidate();
-+         chestHandler = null;
-+      }
++      // NeoForge: Chest state change might change the chest item handler -> invalidate
++      this.invalidateCapabilities();
```

Comment:
In my opinion we should only really invalidate this if `BlockStateProperties.CHEST_TYPE` changed (maybe there is an argument for facing changed as that might make it connect with a different one?). But at the very least it feels off invalidating this when the water logged state changes.

Also the standard for comments was decided as `Neo:` not `NeoForge:`

Reply:
Refined the check a little.